File: 1876. Substrings of Size Three with Distinct Characters/solution.py
class Solution(object):
  def countGoodSubstrings(self, s):
    """
    :type s: str
    :rtype: int
    """
    if len(s) < 3:
      return 0
    
    count = 0
    letters = s[:2]
    for index in range(2, len(s)):
      letters += s[index]
      if letters[0] != letters[1] and letters[1] != letters[2] and letters[2] != letters[0]:
        count += 1
      letters = letters[1:]
      
    return count

File: 1876. Substrings of Size Three with Distinct Characters/solution.js
/**
 * @param {string} s
 * @return {number}
 */
 var countGoodSubstrings = function(s) {
  if (s.length < 3){
    return 0;
  }
  
  let count = 0;
  let letters = s.slice(0, 2);
  for (let index = 2; index < s.length; index++){
    letters += s[index];
    if (letters[0] != letters[1] && letters[1] != letters[2] && letters[2] != letters[0]){
      count += 1;
    }
    letters = letters.slice(1);
  }

  return count;
};